src/interactions/client.ts

```diff
@@ -24,7 +24,8 @@ import { Application } from '../structures/application.ts'
 
 export type ApplicationCommandHandlerCallback = (
   interaction: ApplicationCommandInteraction
-) => unknown
+) => any // Any to include both sync and async return types
```

Comment:
Does `unknown | Promise<unknown>` work here?

Reply:
Yep. In the example I altered you can see the decorators apply to methods both with
https://github.com/harmonyland/harmony/blob/b0d59090358f55ef4f1514d8b408e0975659074b/examples/slash.ts#L84-L92
and without promises
https://github.com/harmonyland/harmony/blob/b0d59090358f55ef4f1514d8b408e0975659074b/examples/slash.ts#L63-L72

Any is more flexible than unknown, although this means less "type secure", so it includes both _Promise\<unknown\>_ and _unknown_. 
[This](https://stackoverflow.com/questions/67798026/restrict-typescript-decorator-to-void-or-promisevoid-functions) could be a solution, but I don't really think that applies here, because a user could return anything from the command callback (maybe to early-exit the function) and I don't really see a point in restricting this freedom.

Anyway If you have a different opinion, feel free to request the change.